### ISSUE TITLE
Allow customization of `console.log`

### ISSUE THREAD

**Issue by lucifer1004** [oven-sh/bun#1686]
### What is the problem this feature would solve?

In Node.JS, users can define a custom `[util.inspect.custom]` method on their classes so that `console.log()` would output using this custom method instead of the default one. This is useful in many cases.

### What is the feature you are proposing to solve the problem?

There should be a way to control the behavior of `console.log()` for a user-defined class.

**Comment by vjpr** [comment]
The newer way (still not implemented btw) is:

```
const customInspectSymbol = Symbol.for('nodejs.util.inspect.custom');

class xxx {
  [customInspectSymbol](depth, inspectOptions, inspect) {
    return `Password <${this.toString()}>`;
  }
}
```

**Comment by Jarred-Sumner** [comment]
@paperdave has landed this in Bun v0.8.0. You can now use `util.inspect.custom` similarly to Node.js.
